web/src/app/login.component.ts

```diff
@@ -104,7 +104,16 @@ export class LoginComponent implements OnInit {
         .joinSession(result.sessionId, result.name)
         .then(() => this.router.navigate(['/']))
         .catch(err => {
-          alert(err);
+          if (err instanceof Error) {
+            alert("Failed to join session: " + err.message);;
+          }
+          else {
+            alert("Failed to join session");
+          }
+          if (err instanceof SessionAlreadyOpenError) {
+            this.router.navigate(['/']);
```

Comment:
If this branch is hit I think the above alert shouldn't occur.

Reply:
I think there should be an alert because the login request of the user is cancelled and the user is redirected to a maybe different session. Redirecting without any notice might be confusing. Anyway, I think there is probably no way, this could ever happen, because login page now also has a guard (see commit: 3d981c5)